dask_ml/decomposition/pca.py

```diff
@@ -210,9 +225,10 @@ def _fit(self, X):
 
         if solver == "auto":
             # Small problem, just call full PCA
-            if max(X.shape) <= 500:
+            n_samples, n_features = dask.compute(X.shape)[0]
```

Comment:
For now, I'd like to avoid any computation that isn't explicitly requested by the user.